www-client/w3m/w3m-0.5.3_p20220216.ebuild

```diff
@@ -0,0 +1,136 @@
+
+LICENSE="w3m"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="fbcon gdk-pixbuf gpm imlib l10n_de l10n_ja lynxkeymap nls nntp ssl unicode X xface"
+
+REQUIRED_USE="
+	fbcon? ( ?? ( gdk-pixbuf imlib ) )
+	X? ( ?? ( gdk-pixbuf imlib ) )
+"
+
+RDEPEND="
+	dev-libs/boehm-gc:=
+	sys-libs/ncurses:0=
+	sys-libs/zlib
+	gdk-pixbuf? (
+		x11-libs/gdk-pixbuf
+		X? ( x11-libs/gdk-pixbuf-xlib )
+	)
+	!gdk-pixbuf? (
+		imlib? ( media-libs/imlib2[X?] )
+	)
+	gpm? ( sys-libs/gpm )
+	ssl? ( dev-libs/openssl:0= )
+	X? (
+		x11-libs/libXdmcp
+		x11-libs/libXext
+	)
+	xface? ( media-libs/compface )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.5.3_p20210102-restore-gettext-macro.patch
+	"${FILESDIR}"/${PN}-0.5.3_p20210102-fix-configure-gettext.patch
+)
+
+src_prepare() {
+	default
+
+	sed -i "/^AR=/s:ar:$(tc-getAR):" {.,${PN}img,libwc}/Makefile.in || die
+	hprefixify acinclude.m4
+
+	AT_NOEAUTOHEADER="yes" AT_M4DIR="m4" eautoreconf
+}
+
+src_configure() {
+	local myconf=()
+	local image imagelib
+
+	if use gdk-pixbuf ; then
+		imagelib="gtk3"
+	elif use imlib ; then
+		imagelib="imlib2"
+	fi
+	if [[ -n "${imagelib}" ]] ; then
+		use X     && image="${image}${image:+,}x11"
+		use fbcon && image="${image}${image:+,}fb"
+	fi
+	# emacs-w3m doesn't like "--enable-m17n --disable-unicode,"
+	# so we better enable or disable both. Default to enable
+	# m17n and unicode, see bug #47046.
+	if use l10n_ja ; then
+		myconf+=( --enable-japanese=$(usex unicode U E) )
+	else
+		myconf+=( --with-charset=$(usex unicode UTF-8 US-ASCII) )
+	fi
+	# lynxkeymap IUSE flag. bug #49397
+	myconf+=( --enable-keymap=$(usex lynxkeymap lynx ${PN}) )
+
+	econf \
+		$(use_enable gpm mouse) \
+		$(use_enable nls) \
+		$(use_enable nntp) \
+		$(use_enable ssl digest-auth) \
+		$(use_enable xface) \
+		$(use_with ssl) \
+		--enable-image=${image:-no} \
+		--enable-m17n \
+		--enable-unicode \
+		--with-browser="${EPREFIX}/usr/bin/xdg-open" \
+		--with-editor="${EPREFIX}/usr/bin/vi" \
+		--with-imagelib="${imagelib:-no}" \
+		--with-mailer="${EPREFIX}/bin/mail" \
+		--without-migemo \
+		--with-termlib=yes \
+		"${myconf[@]}"
+}
+
+src_install() {
+	default
+
+	insinto /etc/${PN}
+	newins "${FILESDIR}"/${PN}.mailcap mailcap
+
+	insinto /usr/share/${PN}/Bonus
+	doins Bonus/{*.{cgi,rb},html*,makeref}
+
+	docinto en
+	dodoc doc/{HISTORY,README,keymap,menu}*
+	newdoc Bonus/README.eng README.Bonus
+	docinto html/en
+	dodoc doc/*.html
+	if use l10n_de ; then
+		docinto de
+		dodoc doc-de/README*
+		docinto html/de
+		dodoc doc-de/*.html
+	else
+		rm -rf "${ED}"/usr/share/man/de || die
+	fi
+	if use l10n_ja ; then
+		docinto ja
+		dodoc doc-jp/{HISTORY,README,keymap,menu}*
+		newdoc Bonus/README README.Bonus
+		docinto html/ja
+		dodoc doc-jp/*.html
+	else
+		rm -rf "${ED}"/usr/share/man/ja || die
+	fi
```

Comment:
We should just always install these files unconditionally and not depend on USE flags. Do you want to look into updating that or should I?
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301

Rest looks good, although there are some things I personally dislike, not worth going into right now.

Reply:
> We should just always install these files unconditionally and not depend on USE flags.

It's done this way on the older ebuild as well, I haven't touched anything not related to version bump.

> Do you want to look into updating that or should I?

Simply removing the if check and leaving the block as is should work right? If so then I can make the changes myself.

Reply:
> Simply removing the if check and leaving the block as is should work right?

Seems to have worked :) @juippis

Reply:
> It's done this way on the older ebuild as well, I haven't touched anything not related to version bump.

I know. Generally when doing version bumps we try to look that things are "top notch". The `rm -f` here alerted me, since that's usually a command that's kept for years even though it hasn't been relevant for a long time. I.e., the files haven't existed for many versions.


> Simply removing the if check and leaving the block as is should work right? If so then I can make the changes myself.

You should also remove the USE flags altogether from the ebuild. I fear there might be some reverse dependency breakage, but haven't looked at that yet.

Reply:
> You should also remove the USE flags altogether from the ebuild.

`ja` is doing something else as well, at line 78. `de` should be safe to remove I guess.